provisionerd/runner/runner.go

```diff
@@ -1051,9 +1055,18 @@ func (r *Runner) runWorkspaceBuild(ctx context.Context) (*proto.CompletedJob, *p
 }
 
 func (r *Runner) failedJobf(format string, args ...interface{}) *proto.FailedJob {
+	message := fmt.Sprintf(format, args...)
+	var code string
+
+	if strings.Contains(message, missingParameterErrorText) {
+		code = MissingParameterErrorCode
+	} else if strings.Contains(message, requiredTemplateVariablesErrorText) {
+		code = RequiredTemplateVariablesErrorCode
+	}
```

Comment:
This format seems like it'd be easy to miss, e.g. when adding new error messages/codes.

How about storing them in a map and iterating over it here?

```go
var errorCodes = map[string]string{
	 MissingParameterErrorCode: missingParameterErrorText,
	// ...
}

for c, m := range errorCodes {
	// ...
}
```

Ultimately though, this logic is a bit hidden behind a "print function", it might be a good idea to refactor so that it can accept an `err error` which could be a predefined `&jobError{m: "missing x", c: "MISSING_X}`.

Reply:
Good idea with the map, added.

>Ultimately though, this logic is a bit hidden behind a "print function", it might be a good idea to refactor so that it can accept an err error which could be a predefined &jobError{m: "missing x", c: "MISSING_X}.

This is risky, as it would require refactoring of interfaces, and I'm sure that I don't want to deal with it now...